redis/tests/test_sentinel.rs

```diff
@@ -0,0 +1,404 @@
+mod support;
+
+use redis::sentinel::SentinelClient;
+
+use crate::support::*;
+
+#[test]
+fn test_sentinel_basics() {
+    let cluster = TestSentinelContext::new(3, 1, 3);
+    let sentinel = cluster.sentinel();
+
+    let mut master_con = sentinel
+        .master_for("master1")
+        .unwrap()
+        .get_connection()
+        .unwrap();
+
+    redis::cmd("SET")
+        .arg("{x}key1")
+        .arg(b"foo")
+        .execute(&mut master_con);
+    redis::cmd("SET")
+        .arg(&["{x}key2", "bar"])
+        .execute(&mut master_con);
+
+    assert_eq!(
+        redis::cmd("MGET")
+            .arg(&["{x}key1", "{x}key2"])
+            .query(&mut master_con),
+        Ok(("foo".to_string(), b"bar".to_vec()))
+    );
+}
+
+#[test]
+fn test_sentinel_read_from_random_replica() {
+    let cluster = TestSentinelContext::new(3, 1, 3);
+    let sentinel = cluster.sentinel();
+
+    let mut master_con = sentinel
+        .master_for("master1")
+        .unwrap()
+        .get_connection()
+        .unwrap();
+
+    let mut replica_con = sentinel
+        .replica_for("master1")
+        .unwrap()
+        .get_connection()
+        .unwrap();
+
+    // Write commands would go to the primary nodes
+    redis::cmd("SET")
+        .arg("{x}key1")
+        .arg(b"foo")
+        .execute(&mut master_con);
+    redis::cmd("SET")
+        .arg(&["{x}key2", "bar"])
+        .execute(&mut master_con);
+
+    // Read commands would go to the replica nodes
+    assert_eq!(
+        redis::cmd("MGET")
+            .arg(&["{x}key1", "{x}key2"])
+            .query(&mut replica_con),
+        Ok(("foo".to_string(), b"bar".to_vec()))
+    );
+}
+
+#[test]
+fn test_sentinel_read_from_multiple_replicas() {
+    let mut cluster = TestSentinelContext::new(3, 1, 3);
+    let sentinel = cluster.sentinel_mut();
+
+    let mut master_con = sentinel
+        .master_for("master1")
+        .unwrap()
+        .get_connection()
+        .unwrap();
+
+    // Write commands would go to the primary nodes
+    redis::cmd("SET")
+        .arg("{x}key1")
+        .arg(b"foo")
+        .execute(&mut master_con);
+    redis::cmd("SET")
+        .arg(&["{x}key2", "bar"])
+        .execute(&mut master_con);
+
+    for _ in 0..20 {
+        let mut replica_con = sentinel
```

Comment:
same, just test that you get a different name on INFO

Reply:
Changing it now.